cumulus/primitives/core/src/lib.rs

```diff
@@ -91,15 +91,30 @@ pub enum AggregateMessageOrigin {
 	///
 	/// This is used by the HRMP queue.
 	Sibling(ParaId),
+	/// This is used by the snowbridge-outbound-queue pallet
+	Export(MessageOrigin),
```

Comment:
I hope we can make this more general somehow. But just so that i understand, this is the outbound queue for bridged messages?  
Will there also be an inbound?

Reply:
> I hope we can make this more general somehow.

I thought about this, came to the conclusion that a generalized solution wouldn't really help, as we can't predict the requirements (or even a need) for the generalized solution. So we could end designing something that will never be used by any projects other than Snowbridge.

But just so that i understand, this is the outbound queue for bridged messages?

Yes. Essentially it takes inbound XCM messages, processes them, and puts them back on the `MessageQueue` for eventual final processing and commitment. Full documentation [here](https://github.com/Snowfork/snowbridge/blob/c9196e196f650e1362dfdf7e911dc9b292cf0859/parachain/pallets/outbound-queue/src/lib.rs#L5). 

> Will there also be an inbound?

Nope. 

> I am not sure if we actually need to change this in the SDK, or if there could be an enum type in the runtime that has this `AggregatedMessageOrigin` as one variant and then a bunch of other custom variants for your snowbridge pallets.

I kinda originally tried something like that in the other old draft PR. Though as you said, it would increase the complexity of runtime upgrades.
```rust
enum AggregateMessageOrigin {
   Xcmp(XcmpOrigin)
   Snowbridge(SnowbridgeOrigin)
}
```

Reply:
Can I see the snowbridge MR to the Polkadot-SDK please? To me it seems like you change this enum but we will never use it in the SDK.  
In that case you can just change it in your snowbridge runtime.

Reply:
Just to clarify, Snowbridge is a set of pallets which will be installed on these system runtimes:
* `bridge-hub-rococo`
* `bridge-hub-kusama`
* `bridge-hub-polkadot`

So within those runtimes, the XCMP and Snowbridge subsystems will need to agree on a common `AggregateMessageOrigin`, unless I'm missing something:

For example here we install the pallets in the `bridge-hub-rococo` runtime: https://github.com/Snowfork/polkadot-sdk/blob/71197605d792abca283c01c5873bfe8e1dba83c8/cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/lib.rs#L681-L684

Here we configure the `MessageQueue` to use our `OutboundQueue` as a message processer: https://github.com/Snowfork/polkadot-sdk/blob/71197605d792abca283c01c5873bfe8e1dba83c8/cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/lib.rs#L493-L495

And here's our interim `AggregateMessageOrigin`: 
https://github.com/Snowfork/snowbridge/blob/c9196e196f650e1362dfdf7e911dc9b292cf0859/parachain/primitives/core/src/outbound.rs#L364


Is hard to give you a clean MR right now, as the base branch for our working branch is a bit of frankenstein at the moment.

Reply:
But then there should maybe a uniform `AggregateMessageOrigin` across all `bridge-hub` runtimes. So it should be in a `bridge-hub-common` crate or something. It does not make sense to force this onto parachain that will never use it.

Reply:
That makes very good sense, narrowing the change down to the BridgeHub level. Let me try that.